skyportal/handlers/api/spectrum.py

```diff
@@ -289,27 +245,21 @@ def delete(self, spectrum_id):
               application/json:
                 schema: Error
         """
-        spectrum = Spectrum.query.get(spectrum_id)
-        # Permissions check
-        _ = Obj.get_if_readable_by(spectrum.obj_id, self.current_user)
-
-        # Check that the requesting user owns the spectrum (or is an admin)
-        if not spectrum.is_modifiable_by(self.associated_user_object):
-            return self.error(
-                f'Cannot delete spectrum that is owned by {spectrum.owner}.'
-            )
-
+        spectrum = Spectrum.get_if_accessible_by(
+            spectrum_id, self.current_user, mode="delete", raise_if_none=True
+        )
+        obj_key = spectrum.obj.internal_key
         DBSession().delete(spectrum)
         self.verify_and_commit()
 
         self.push_all(
             action='skyportal/REFRESH_SOURCE',
-            payload={'obj_key': spectrum.obj.internal_key},
+            payload={'obj_key': obj_key},
         )
 
         self.push_all(
             action='skyportal/REFRESH_SOURCE_SPECTRA',
-            payload={'obj_id': spectrum.obj_id},
+            payload={'obj_id': obj_key},
```

Comment:
I think this should still be spectrum.obj_id - the internal_key and obj_id are different things, and the spectra duck still expects the obj_id here

Reply:
Agreed, good catch

Reply:
made this change, thanks @kmshin1397 .